common/src/main/java/com/microsoft/identity/common/internal/ui/webview/AzureActiveDirectoryWebViewClient.java

```diff
@@ -160,6 +171,36 @@ private boolean handleUrl(final WebView view, final String url) {
                 final PKeyAuthChallenge pKeyAuthChallenge = factory.getPKeyAuthChallengeFromWebViewRedirect(url);
                 final PKeyAuthChallengeHandler pKeyAuthChallengeHandler = new PKeyAuthChallengeHandler(view, getCompletionCallback());
                 pKeyAuthChallengeHandler.processChallenge(pKeyAuthChallenge);
+            } else if (FidoConstants.IS_PASSKEY_SUPPORT_READY && isPasskeyUrl(formattedURL)) {
+                Logger.info(methodTag,"WebView detected request for passkey protocol.");
+                final SpanContext spanContext = getActivity() instanceof AuthorizationActivity ? ((AuthorizationActivity)getActivity()).getSpanContext() : null;
+                final PasskeyFidoChallengeHandler challengeHandler = new PasskeyFidoChallengeHandler(
+                        new CredManFidoManager(view.getContext()),
+                        view,
+                        spanContext,
+                        ViewTreeLifecycleOwner.get(view));
+                try {
+                    final IFidoChallenge challenge = FidoChallengeFactory.createFidoChallengeFromRedirect(url);
+                    challengeHandler.processChallenge(challenge);
+                } catch (final ClientException e) {
+                    // Server wants to know if there's an incorrect query string parameter field/name sent by them,
+                    // so we need to validate and get submitUrl and context first in order to send those related exceptions to them.
+                    // If the submitUrl or context field themselves are incorrect... another ClientException will be thrown and the WebView will close.
+                    final Map<String, String> parameters = UrlUtil.getParameters(url);
```

Comment:
I'm wondering if these check should be in PasskeyFidoChallengeHandler instead.

Reply:
If I assume that the challenge content might greatly vary (a required field in Auth flow might not be required in Reg flow, and vice versa), having its own flow validate required value might make more sense.

If that's the case, I could even have a challenge class that simply takes in the url as-is. The param construction and validation can be done separately in each flow.

You would need a factory for PasskeyFidoChallengeHandler instead (to determine if it's auth or reg, etc. would there ever be a hint for that?).

Reply:
So a general FidoChallenge class that has all the parameters combined from reg and auth (maybe just auth for now, since reg will come much later)? 
Then the challenge handler factory will determine which handler to return (this would be tricky without some hint from the protocol, as some fields can be malformed, so that's something I would ask the server side to provide. for now, it'll just return auth without checks. )
Then the challenge handler would do verification (not null, not empty, valid submiturl)

Iiuc, then I'll move the verification stuff from the challenge factory to the challenge handler itself.

Reply:
Ended up doing some refactoring:
- Created FidoChallenge, which breaks down the uri but does no validation
- moved validation to AuthFidoChallengeHandler
- Moved and rewrote a lot of tests

I'm not sure if a factory for FidoChallengeHandler is needed right now, but if it would be good practice to have one, I can add it in a different PR (to avoid further increasing the size of this one)